SDL/NeuralNetwork.cuh

```diff
@@ -0,0 +1,188 @@
+    CUDA_CONST_VAR const float LSTWP1 = 0.3418833f; // 94.0% TPR, 43.9% FPR
+    CUDA_CONST_VAR const float LSTWP2 = 0.6177366f; // 82.0% TPR, 20.0% FPR
+    // Other working points
+    CUDA_CONST_VAR const float WP70   = 0.7776195f; // 70.0% TPR, 10.0% FPR
+    CUDA_CONST_VAR const float WP75   = 0.7181118f; // 75.0% TPR, 13.5% FPR
+    CUDA_CONST_VAR const float WP80   = 0.6492643f; // 80.0% TPR, 17.9% FPR
+    CUDA_CONST_VAR const float WP85   = 0.5655319f; // 85.0% TPR, 23.8% FPR
+    CUDA_CONST_VAR const float WP90   = 0.4592205f; // 90.0% TPR, 32.6% FPR
+    CUDA_CONST_VAR const float WP95   = 0.3073708f; // 95.0% TPR, 47.7% FPR
+    CUDA_CONST_VAR const float WP97p5 = 0.2001348f; // 97.5% TPR, 61.2% FPR
+    CUDA_CONST_VAR const float WP99   = 0.1120605f; // 99.0% TPR, 75.9% FPR
+    CUDA_CONST_VAR const float WP99p9 = 0.0218196f; // 99.9% TPR, 95.4% FPR
+
+    CUDA_DEV inline float runInference(struct SDL::modules& modulesInGPU, struct SDL::miniDoublets& mdsInGPU, 
+                                       struct SDL::segments& segmentsInGPU, struct SDL::triplets& tripletsInGPU, 
+                                       float* xVec, float* yVec, unsigned int* mdIndices, const uint16_t* lowerModuleIndices, 
+                                       unsigned int& innerTripletIndex, unsigned int& outerTripletIndex, 
+                                       float& innerRadius, float& outerRadius, float& bridgeRadius)
+    {
+        // Unpack x-coordinates of hits
+        float x1 = xVec[0];
+        float x2 = xVec[1];
+        float x3 = xVec[2];
+        float x4 = xVec[3];
+        float x5 = xVec[4];
+        // Unpack y-coordinates of hits
+        float y1 = yVec[0];
+        float y2 = yVec[1];
+        float y3 = yVec[2];
+        float y4 = yVec[3];
+        float y5 = yVec[4];
+        // Unpack module indices
+        unsigned int mdIndex1 = mdIndices[0];
+        unsigned int mdIndex2 = mdIndices[1];
+        unsigned int mdIndex3 = mdIndices[2];
+        unsigned int mdIndex4 = mdIndices[3];
+        unsigned int mdIndex5 = mdIndices[4];
+        // Unpack module indices
+        uint16_t lowerModuleIndex1 = lowerModuleIndices[0];
+        uint16_t lowerModuleIndex2 = lowerModuleIndices[1];
+        uint16_t lowerModuleIndex3 = lowerModuleIndices[2];
+        uint16_t lowerModuleIndex4 = lowerModuleIndices[3];
+        uint16_t lowerModuleIndex5 = lowerModuleIndices[4];
+
+        // Compute some convenience variables
+        short layer2_adjustment = 0;
+        if (modulesInGPU.layers[lowerModuleIndex1] == 1)
+        {
+            layer2_adjustment = 1; // get upper segment to be in second layer
+        }
+        unsigned int md_idx_for_t5_eta_phi = segmentsInGPU.mdIndices[2*tripletsInGPU.segmentIndices[2*innerTripletIndex + layer2_adjustment]];
+        bool is_endcap1 = (modulesInGPU.subdets[lowerModuleIndex1] == 4);                // true if anchor hit 1 is in the endcap
```

Comment:
I don't see `is_endcap*` used by itself; it's unclear if the explicit temporary variable is useful here.

Reply:
better check that the compiler optimizes these away, so that we don't pay a price of an extra register just to define a temporary with a shorter name